.github/workflows/discord-notification.yml

```
@@ -0,0 +1,107 @@
+name: Discord Notification
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize, closed]
+  push:
+    branches: [main, develop] # Add your target branches
+  issues:
+    types: [opened, closed, reopened]
+  workflow_run:
+    workflows: ["CI", "Tests"] # Add your other workflow names
+    types: [completed]
+
+jobs:
+  notify:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get latest commit message
+        id: commit
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const { data: commits } = await github.rest.pulls.listCommits({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: context.payload.pull_request.number
+            });
+            const lastCommit = commits[commits.length - 1];
+            core.setOutput('message', lastCommit.commit.message);
+
+      - name: Discord PR Notification
+        if: github.event_name == 'pull_request'
+        env:
+          DISCORD_WEBHOOK: ${{ secrets.DISCORD_WEBHOOK }}
+        uses: Ilshidur/action-discord@master
+        with:
+          args: |
+            ${{ github.event.action == 'opened' && '🆕 **New Pull Request**' || '🔄 **Pull Request Updated**' }}
+
+            **Latest Commit:** ${{ steps.commit.outputs.message }}
+
+            **Description:** 📝
+            ${{ github.event.pull_request.body }}
+
+            🔗 **Link:** ${{ github.event.pull_request.html_url }}
+
+      - name: Discord PR Merged Notification
+        if: github.event_name == 'pull_request' && github.event.pull_request.merged == true
+        env:
+          DISCORD_WEBHOOK: ${{ secrets.DISCORD_WEBHOOK }}
+        uses: Ilshidur/action-discord@master
+        with:
+          args: |
+            **Pull Request Merged! 🎉**
+
+            **Title:** ${{ github.event.pull_request.title }}
+            **Author:** ${{ github.event.pull_request.user.login }}
+            **Merged by:** ${{ github.event.pull_request.merged_by.login }}
+
+            🔗 **Link:** ${{ github.event.pull_request.html_url }}
+
+      - name: Discord Push Notification
+        if: github.event_name == 'push'
```

```suggestion
        if: github.event_name == 'push' && !contains(github.event.head_commit.message, '[skip notification]')
```
Added conditional logic to check commit messages for [skip notification] tag

Developers can now control notification flow based on commit type
Reduces noise from minor changes (docs, formatting, dependencies)